py/examples/file_reader_gps/main.py

```diff
@@ -0,0 +1,150 @@
+        tuple: The unpacked message.
+    """
+
+    if msg_type == "relposned":
+
+        stamp: float = msg.stamp.stamp
+        gps_time: float = msg.gps_time.stamp
+        relative_pose_north: float = msg.relative_pose_north
+        relative_pose_east: float = msg.relative_pose_east
+        relative_pose_down: float = msg.relative_pose_down
+        relative_pose_length: float = msg.relative_pose_length
+        accuracy_north: float = msg.accuracy_north
+        accuracy_east: float = msg.accuracy_east
+        accuracy_down: float = msg.accuracy_down
+        carr_soln: int = msg.carr_soln
+        gnss_fix_ok: bool = msg.gnss_fix_ok
+
+        gps_message = (
+            stamp,
+            gps_time,
+            relative_pose_north,
+            relative_pose_east,
+            relative_pose_down,
+            relative_pose_length,
+            accuracy_north,
+            accuracy_east,
+            accuracy_down,
+            carr_soln,
+            gnss_fix_ok,
+        )
+
+    else:
+        stamp: float = msg.stamp.stamp
+        gps_time: float = msg.gps_time.stamp
+        latitude: float = msg.latitude
+        longitude: float = msg.longitude
+        altitude: float = msg.altitude
+        ground_speed: float = msg.ground_speed
+        speed_accuracy: float = msg.speed_accuracy
+        horizontal_accuracy: float = msg.horizontal_accuracy
+        vertical_accuracy: float = msg.vertical_accuracy
+        p_dop: float = msg.p_dop
+
+        gps_message = (
+            stamp,
+            gps_time,
+            latitude,
+            longitude,
+            altitude,
+            ground_speed,
+            speed_accuracy,
+            horizontal_accuracy,
+            vertical_accuracy,
+            p_dop,
+        )
+
+    return gps_message
+
+
+def main(file_name: str, msg_type: str) -> None:
+    if msg_type not in ["relposned", "pvt"]:
+        raise RuntimeError(f"Message type not recognized: {msg_type}")
+
+    # create the file reader
+    reader = EventsFileReader(file_name)
+    success: bool = reader.open()
+    if not success:
+        raise RuntimeError(f"Failed to open events file: {file_name}")
+
+    # get the index of the events file
+    events_index: list[EventLogPosition] = reader.get_index()
+
+    # structure the index as a dictionary of lists of events
+    events_dict: dict[str, list[EventLogPosition]] = build_events_dict(events_index)
+    print(f"All available topics: {sorted(events_dict.keys())}")
+
+    gps_events = events_dict[f"/gps/{msg_type}"]
+    print(f"Found {len(gps_events)} packets of gps/{msg_type}\n")
+
+    for event_log in gps_events:
+
+        # parse the message
+        msg: gps_pb2.RelativePositionFrame | gps_pb2.GpsFrame = event_log.read_message()
+        gps_msg = unpack_gps_message(msg, msg_type)
+
+        if msg_type == "pvt":
+            print(f"Message stamp: {gps_msg[0]}")
+            print(f"GPS time: {gps_msg[1]}")
+            print(f"Latitude: {gps_msg[2]}")
```

Comment:
I think you should remove the function that puts it into a tuple and just print the values directly from the proto. It'd be different if you were creating a class or something, but this just adds risk of indexing error without adding any benefit
```suggestion
            print(f"Message stamp: {msg.stamp.stamp}")
            print(f"GPS time: {msg.gps_time.stamp}")
            print(f"Latitude: {msg.latitude}")
            ...
```

Reply:
Done in commit 31429d56be911d2fc592f2de6730cfa3dfd9ad5c